components/hooks/useSavingIndicator.ts

```
@@ -0,0 +1,35 @@
+
+export function useSavingIndicator() {
+  const [state, setState] = useState<SavingState>("idle");
+
+  const setSaving = () => setState("saving");
+  
+  const setSaved = () => {
+    setState("saved");
+    // Auto-hide the "saved" indicator after 2 seconds
+    setTimeout(() => setState("idle"), 2000);
+  };
+  
```

ESLint

[error] Delete `··`

(prettier/prettier)